### PR TITLE
Improve error messages on use of contract ids in contract keys

### PR DESCRIPTION
**Pull request by cocreature**
We got some feedback that "Unexpected contract id" sounds like an
internal error whereas this is really a user error.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by remyhaemmerle-da — APPROVED** [review]
Thanks.

**Review by hurryabit — APPROVED** [review]
Good idea. Thank you.

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala [hunk 1]
```diff
@@ -94,7 +94,7 @@ private[lf] final class CommandPreprocessor(compiledPackages: CompiledPackages)
     val (arg, argCids) = valueTranslator.unsafeTranslateValue(choiceArgType, argument)
     val (key, keyCids) = valueTranslator.unsafeTranslateValue(ckTtype, contractKey)
     keyCids.foreach { coid =>
-      fail(s"Unexpected contract id in key: $coid")
+      fail(s"Contract IDs are not supported in contract keys: $coid")
```
Comment (gerolf-da) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala:97]: Would it make sense to include the templateId as well, or will this be clear in the context of the error message?

Comment (cocreature) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/CommandPreprocessor.scala]: sounds sensible, added.

**Pull request merged**